src/core/localize.rkt

```diff
@@ -191,73 +191,174 @@
 ;; same shape as `prog`
 (define (local-error-as-tree test ctx)
-  (define errs (first (compute-local-errors (list (all-subexpressions (test-input test))) ctx)))
-
-  (define local-error
-    (let loop ([expr (test-input test)])
-      (define expr-info (hash-ref errs expr))
-      (define err-list (hash-ref expr-info 'errs))
-      (match expr
-        [(list op args ...) (cons err-list (map loop args))]
-        [_ (list err-list)])))
-
-  (define exact-values
-    (let loop ([expr (test-input test)])
-      (define expr-info (hash-ref errs expr))
-      (define exacts-list (hash-ref expr-info 'exact-values))
-      (match expr
-        [(list op args ...) (cons exacts-list (map loop args))]
-        [_ (list exacts-list)])))
-
-  (define actual-values
-    (let loop ([expr (test-input test)])
-      (define expr-info (hash-ref errs expr))
-      (define actual-list (hash-ref expr-info 'actual-values))
-      (match expr
-        [(list op args ...) (cons actual-list (map loop args))]
-        [_ (list actual-list)])))
-
-  (define tree
-    (let loop ([expr (prog->fpcore (test-input test) (test-context test))]
-               [err local-error]
-               [exact exact-values]
-               [actual actual-values])
-      (match expr
-        [(list op args ...)
-         ;; err => (List (listof Integer) List ...)
-         (hasheq 'e
-                 (~a op)
-                 'avg-error
-                 (format-bits (errors-score (first err)))
-                 'exact-value
-                 (map ~s (first exact))
-                 'actual-value
-                 (map ~s (first actual))
-                 'children
-                 (map loop args (rest err) (rest exact) (rest actual)))]
-        ;; err => (List (listof Integer))
-        [_
-         (hasheq 'e
-                 (~a expr)
-                 'avg-error
-                 (format-bits (errors-score (first err)))
-                 'exact-value
-                 (map ~s (first exact))
-                 'actual-value
-                 (map ~s (first actual))
-                 'children
-                 '())])))
-  tree)
+  (define fpcore (prog->fpcore (test-input test) (test-context test)))
+  (define subexprss (list (all-subexpressions (test-input test))))
+  (define exprs-list (append* subexprss)) ; unroll subexprss
+  (define ctx-list
+    (for/list ([subexpr (in-list exprs-list)])
+      (struct-copy context ctx [repr (repr-of subexpr ctx)])))
+  (define repr-hash
+    (make-immutable-hash (map (lambda (e ctx) (cons e (context-repr ctx))) exprs-list ctx-list)))
+
+  (define ctx-vec (list->vector ctx-list))
+  (define spec-list (map prog->spec exprs-list))
+  (define spec-vec (list->vector spec-list))
+  (define expr-batch (progs->batch exprs-list))
+  (define nodes (batch-nodes expr-batch))
+  (define roots (batch-roots expr-batch))
+
+  ; TODO don't ignore the status code from make-real-compiler in eval-progs-real
```

Comment:
I don't think `make-real-compiler` has a status code, it's evaluation that has a status code.